src/main/java/frc/robot/subsystems/conveyor/ConveyorIOReal.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.subsystems.conveyor;
+
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Velocity;
+import frc.robot.Constants;
+
+public class ConveyorIOReal implements ConveyorIO {
+
+    private CANSparkMax roller = new CANSparkMax(1, CANSparkLowLevel.MotorType.kBrushless);
+
+    public ConveyorIOReal() {
+        roller.restoreFactoryDefaults();
+        roller.setIdleMode(CANSparkMax.IdleMode.kCoast);
+        roller.enableVoltageCompensation(Constants.NOMINAL_VOLTAGE);
+        roller.setInverted(true);
+        for (int i = 1; i <= 6; i++) {
+            roller.setPeriodicFramePeriod(CANSparkLowLevel.PeriodicFrame.fromId(i), 50_000);
+        }
+    }
+
+
+    @Override
+    public void setVelocity(MutableMeasure<Velocity<Angle>> velocity) {
+        roller.set(velocity.in(Units.RotationsPerSecond));
+    }
+
+    @Override
+    public void updateInputs() {
+        inputs.currentVelocity.mut_replace(Units.RotationsPerSecond.of(roller.getEncoder().getVelocity()).mutableCopy());
```

Comment:
I'm not sure, do you have to call `mutableCopy` everywhere?

Reply:
no.

Reply:
fe653a428065d4ed023b41a4f1444dcdcf39c64f